source/common/http/filter_manager.cc

```diff
@@ -277,6 +277,48 @@ void ActiveStreamFilterBase::resetIdleTimer() {
   parent_.filter_manager_callbacks_.resetIdleTimer();
 }
 
+const Router::RouteSpecificFilterConfig*
+ActiveStreamFilterBase::mostSpecificPerFilterConfig() const {
+  ASSERT(base_filter_ != nullptr);
+  auto route = parent_.filter_manager_callbacks_.route(nullptr);
+  if (route == nullptr) {
+    return nullptr;
+  }
+
+  std::string custom_name(base_filter_->customName());
+  auto* result = route->mostSpecificPerFilterConfig(custom_name);
+
+  if (result == nullptr) {
```

Comment:
We can skip the fallback if the filter doesn't use the custom name, so the check can be:
`if (result == nullptr && base_filter_->customName() != base_filter_->fileName())`